### PR TITLE
Control the intregration slider using arrow keys

### LINKED ISSUE #28342: Make the TOF slider arrow-controlled
In the instrument view, it should be possible to control the time-of-flight slider using the arrow keyboard keys, for convenience sake.

### PR DESCRIPTION
**Pull request by MathieuTillet**
**Description of work.**
This PR allows to control the integration slider, in the instrument view, using the left and right arrow keys.

**To test:**
- On workbench, load a non-monochromatic file, such as `ExternalData/Testing/Data/UnitTest/ILL/D17/397806.nxs`.
- Open the instrument window, and play with the integration bar. In particular, check the view is only updated when the key is released.
 

Fixes #28342. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by Harrietbrown — CHANGES_REQUESTED** [review]
Very nice, the keyborad controls work quite well, while we are adding left and right control could we also have control over the slider size wth the keyboard, perhaps with the up and down arrow?

**Comment by MathieuTillet** [comment]
Good idea, I'll add that. In this case, I'll also try removing the lower size limit of the slider, so the user can examine one channel at a time if he wants.

File: qt/widgets/instrumentview/src/XIntegrationControl.cpp
Comment (Harrietbrown) [inline qt/widgets/instrumentview/src/XIntegrationControl.cpp:161]: All these if statements are a bit hard to look through, can you put in some comments on what each catch is checking for?

Comment (MathieuTillet) [reply qt/widgets/instrumentview/src/XIntegrationControl.cpp]: Well, they were so confusing, one of them was actually redundant. It should be easier to understand now.

**Review by Harrietbrown — DISMISSED** [review]
The comments so a good job explaining what the if statement tree does and the slider bar functions as described, Good job.

File: qt/widgets/instrumentview/src/XIntegrationControl.cpp
Comment (gvardany) [inline qt/widgets/instrumentview/src/XIntegrationControl.cpp:214]: This  comment seems to be a leftover, can you remove it?

Comment (MathieuTillet) [reply qt/widgets/instrumentview/src/XIntegrationControl.cpp]: Done

**Comment by MathieuTillet** [comment]
Just fixing a conflict in the release notes.

**Comment by gvardany** [comment]
@MathieuTillet Annoyingly there is a new conflict now. @Harrietbrown could you approve again once that one is also merged?

**Review by Harrietbrown — APPROVED** [review]
The changes still work after the rebase good job.

**Pull request merged**
